### ISSUE TITLE
Support custom field to store extra info in JSON message

### ISSUE THREAD

**Issue by nminhquan** [zendesk/maxwell#913]
Hi folks,

Is it possible that Maxwell support adding custom fields (to store extra info) into the generated JSON message, like:
{
    "database": "test",
    "table": "maxwell",
    "type": "update",
    "ts": 1449786341,
    "xid": 940786,
    "commit": true,
    **"custom_field1": "1",**
    "data": {"id":1, "daemon": "Firebus!  Firebus!"},
    "old":  {"daemon": "Stanislaw Lem"}
  }

Many thanks!

**Comment by osheroff** [comment]
Currently this is possible with a custom producer and RowMap#putExtraAttribute.  When #906 lands this will also be accessible from a JavaScript filter.

**Comment by nminhquan** [comment]
Thanks, @osheroff 
Actually, my intention was to reuse the built-in Kafka Producer but wanted to add an extra field of information into it. Is it possible?
As a quick look, I tried writing a custom producer which will work as-is the built-in Kafka Producer but I got stuck with the constructor, where the ProducerFactory just pass the(MaxwellContext object, while Kafka Producer requires more than that (MaxwellContext context, Properties kafkaProperties, String kafkaTopic)

Thanks in advance.
